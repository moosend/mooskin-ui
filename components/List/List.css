@import url('https://fonts.googleapis.com/css?family=Alegreya+Sans:500|Convergence');
@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';

.list{
    display: flex;
    flex-direction: column;
    background: #fff;
    border-top: 1px solid #efefef;
    border-bottom: 1px solid #efefef;
}

.listItem{
    /* display: flex; */
    /* flex-direction: row; */
    position: relative;
}

.listItem + .listItem {
    border-top: 1px solid #efefef;
}

/* .image{
    display: flex;
    flex-grow: 1;
    flex-shrink: 1;
    justify-content: center;
    align-items: center;
    overflow: hidden
}

.image img{
    flex-shrink: 0;
    flex-grow: 0; 
    min-width: 100%;
    max-width: 100%;   
}

.titleContainer{
    display: flex;
    flex: 4;
    flex-direction: column;
    justify-content: space-around;
    padding-left: 20px;
}

.detailsContainer{
    display: flex;
    flex-grow: 1;
    flex-shrink: 1;
    flex-direction: row;
}

.title{
    font-family: 'Convergence', 'Alegreya Sans', sans-serif;
    font-size: 1.25rem;
}

.description{
    color: var(--italicDescription);
    font-size: .78571428571rem;
} */

.content:hover {
    background: #F6F6F6;
}

.expandedSection {
    background: #f9f9f9;
    border-top: 1px solid #eaeaea;
    border-bottom: 1px solid #eaeaea;
    transition: all 0.4s;
    position: relative;
    /* overflow: hidden; */
    /* height: 1px; */
}

.arrow {
    top: 1px;
    left: 20px;
    z-index: 5;
    position: absolute;
}

.innerArrow{
    width: 0px;
    height: 0px;
    border-left: 20px solid transparent;
    border-right: 20px solid transparent;
    border-bottom: 20px solid #eaeaea;
    /* border-bottom: 20px solid black; */
    position: absolute;
    bottom: 0px;
}

.borderArrow{
    width: 0;
    height: 0;
    border-left: 19px solid transparent;
    border-right: 19px solid transparent;
    border-bottom: 19px solid #f9f9f9;
    z-index: 11;
    position: absolute;
    bottom: 0px;
    left: 1px;
}

@media(--xSmall){

    /* .listItem{
        flex-direction: column;
    } */

    .detailsContainer{
        margin-bottom: 15px;
    }

    .titleContainer{
        flex: 3;
    }
}