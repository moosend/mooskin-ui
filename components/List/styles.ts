import styled from 'styled-components';

// Models
import { IBoxComponentProps } from '../Box/model';

// Components
import { Box } from '../Box/Box';

// "CSS" variables
import variables from '../_utils/globals/variables';

export const StyledListItem = styled(Box)<IBoxComponentProps>`
	display: flex;
	padding: 15px;
	background-color: ${(props) => props.palette?.backgroundColors.white || variables.backgroundColors.white};
	color: ${(props) => props.palette?.fontColors.text || variables.fontColors.text};
`;

StyledListItem.displayName = 'StyledListItem';

export const StyledListItemCommon = styled(Box)<IBoxComponentProps>`
	display: flex;
	color: inherit;
`;

StyledListItemCommon.displayName = 'StyledListItemCommon';

export const StyledListItemHead = styled(StyledListItemCommon)``;

StyledListItemHead.displayName = 'StyledListItemHead';

export const StyledListItemBody = styled(StyledListItemCommon)`
	flex: 1;
	flex-direction: column;
	padding: 0 20px;
`;

StyledListItemBody.displayName = 'StyledListItemBody';

export const StyledListItemEnd = styled(StyledListItemCommon)`
	align-items: center;
`;

StyledListItemEnd.displayName = 'StyledListItemEnd';
