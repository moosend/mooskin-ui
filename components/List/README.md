# MooSkin UI - List & ListItem Component

The MooSkin List & ListItem Component works similar to the normal HTML `<ol/>`, `<ul/>`, `<li/>` elements, but with pre-defined styling and attributes.

___

### Usage

To start using the List & ListItem Component first you have to Import it

```
Import {List, ListItem, ItemContent} from 'mooskin';
```
or modular import
```
// JS
import List, {ListItem} from 'mooskin/lib/List';

// CSS
import 'mooskin/lib/List/style.css';
```

And then you can simply start using it by typing

```
<List attribute1="atr" attribute2="asd" >
    <ListItem>
        <ItemContent>
            content 1
        <ItemContent/>
    </ListItem>
    <ListItem>
        <ItemContent>
            content 2
        <ItemContent/>
    </ListItem>
    <ListItem>
        <ItemContent>
            content 3
        <ItemContent/>
    </ListItem>
</List>
```

Like normal html elements, it will accept given attributes and render differently based on the given attributes

### Examples

Give it a custom class or style

```
<List style={yourStyle} className={yourClass} >...</List>
```

or pass props to the LinkItem component

```
<List>
    <ListItem style={yourStyle} className={yourClass}>...</ListItem>
</List>
```

### ExpandedSection content

A list item can have an expandable hidden content which will reveal when the `ListItem` prop of `expanded` is true, example:

```
Import {List, ListItem, ItemContent, ExpandedSection} from 'mooskin';

<List attribute1="atr" attribute2="asd" >
    <ListItem>
        <ItemContent>
            content 1
        <ItemContent/>
        <ExpandedSection expanded={true || false}>
            expandable content 1
        <ExpandedSection/>
    </ListItem>
    <ListItem>
        <ItemContent>
            content 2
        <ItemContent/>
        <ExpandedSection expanded={true || false}>
            expandable content 2
        <ExpandedSection/>
    </ListItem>
    <ListItem expanded>
        <ItemContent>
            content 3
        <ItemContent/>
        <ExpandedSection expanded={true || false}>
            expandable content 3
        <ExpandedSection/>
    </ListItem>
</List>
```
so in the above example, the third and last list item will have the hidden section expanded, the others will have the section hidden.

<div class="playground-doc">

## Supported List attributes

* `id` - id of the element
* `style` - listitem style
* `className` - css class

## Supported ListItem attributes

* `image` - image related to the list item
* `title` - list item title
* `expanded` - expands the hidden section of the list item
* `description` - list item description
* `className` - css class
* `style` - listitem style

## Supported ListContent attributes

* `id` - id of the element
* `style` - listitem style
* `className` - css class

## Supported ExpandedSection attributes

* `id` - id of the element
* `style` - listitem style
* `className` - css class
* `expanded` - if true, displays hidden content
* `arrow` - boolean value wether to display expanded section arrow, defaults to true

</div>

Allthough these attributes are supported, all of them are optional.


#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)