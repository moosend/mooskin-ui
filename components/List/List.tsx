import * as React from 'react';

import styles from './List.css';

export interface IListProps {

    /** list id */
    id?: string;

    /** list class */
    className?: string;

    /** color to be applied when an item is hovered */
    // hoverColor?: string;

    /** override styles */
    style?: React.CSSProperties;

    /** children here can only be Tab elements */
    children?: React.ReactElement<IListItemProps> | Array<React.ReactElement<IListItemProps>> | JSX.Element | Element;

}

export interface IListItemProps {

    /** wether the list item is expanded */
    // expanded?: boolean;

    /** listitem image */
    // image?: string;

    /** listitem title */
    // title?: string;

    /** listitem description */
    // description?: string;

    /** inherited from parent */
    // hovered?: boolean;

    /** color to be applied when this item is hovered (inherited by default) */
    // hoverColor?: string;

    /** listitem class */
    className?: string;

    /** override styles */
    style?: React.CSSProperties;

    /** listitem children */
    children?: React.ReactElement<IExpandedSectionProps> | React.ReactElement<IItemContentProps> |
        Array<React.ReactElement<IExpandedSectionProps> | React.ReactElement<IItemContentProps>>;
    // children?: any;

    // onMouseEnter?: (e: React.MouseEvent<HTMLElement>) => void;

    // onMouseLeave?: (e: React.MouseEvent<HTMLElement>) => void;

    onClick?: (e: React.MouseEvent<HTMLElement>) => void;

}

// export interface IListState {
//     hoverItem?: number;
//     expandedLists: number[];
// }

export default class List extends React.Component<IListProps, {} /* IListState */>{

    static defaultProps = {
        className: '',
        style: {}
    };

    constructor(props: IListProps){
        super(props);

        // this.state = {
        //     expandedLists: [],
        //     hoverItem: -1
        // };
    }

    render(){

        const classes = `${styles.list} ${this.props.className}`;

        return(
            <div className={`list-component ${classes}`} style={this.props.style} id={this.props.id}>
                {/* {this.getItems()} */}
                {this.props.children}
            </div>
        );
    }

    // getItems = () => {
    //     const items: Array<React.ReactElement<IListItemProps>> = [];
    //     React.Children.forEach(this.props.children, (child, index) => {
    //         React.isValidElement<IListItemProps>(child) &&
    //                 items.push(this.renderListItemStructure(child, index));
    //             // const onClick = expandedSection ? this.onListClick(index) : undefined;
    //             // child.props.expanded && this.setActiveState(index);
    //             // const expanded = child.props.expanded ? child.props.expanded
    //             //             : this.state.expandedLists.includes(index) ? true : false;
    //             // const cursor = expandedSection ? {cursor: 'pointer'} : {};
    //     });
    //     return items;
    // }

    // renderListItemStructure = (child: React.ReactElement<IListItemProps>, index: number) => {
    //     return (
    //         <ListItem
    //             key={index}
    //             // expanded={child.props.expanded}
    //             hovered={index === this.state.hoverItem}
    //             className={child.props.className}
    //             // image={child.props.image}
    //             // description={child.props.description}
    //             // title={child.props.title}
    //             style={child.props.style}
    //             hoverColor={child.props.hoverColor || this.props.hoverColor}
    //             onMouseEnter={this.onItemEnter(index)}
    //             onMouseLeave={this.onItemLeave}
    //             onClick={child.props.onClick}
    //         >
    //             {child.props.children}
    //         </ListItem>
    //     );
    // }

    // renderListItemStructure = (child: React.ReactElement<IListItemProps>, index: number) => {
    //     if (Array.isArray(child.props.children) && child.props.children.length > 1 && child.type === ListItem){
    //         const {content, expandedSection} =
    //                 this.getItemChildren(child.props.children, child.props.expanded || false, index);
    //         if (child.props.expanded && !expandedSection){
    //             throw new Error('No expanded section available to be expanded!');
    //         }
    //         return (
    //             <div style={{display: 'flex', flexDirection: 'column'}} key={index}>
    //                 <ListItem
    //                     expanded={child.props.expanded}
    //                     hovered={index === this.state.hoverItem}
    //                     className={child.props.className}
    //                     // image={child.props.image}
    //                     // description={child.props.description}
    //                     // title={child.props.title}
    //                     style={child.props.style}
    //                     hoverColor={child.props.hoverColor || this.props.hoverColor}
    //                     onMouseEnter={this.onItemEnter(index)}
    //                     onMouseLeave={this.onItemLeave}
    //                     onClick={child.props.onClick}
    //                 >
    //                     {content}
    //                 </ListItem>
    //                 {expandedSection}
    //             </div>
    //         );
    //     }
    //     if (child.type !== ListItem){
    //         return (
    //             <ListItem
    //                 expanded={child.props.expanded}
    //                 hovered={index === this.state.hoverItem}
    //                 className={child.props.className}
    //                 // image={child.props.image}
    //                 // description={child.props.description}
    //                 // title={child.props.title}
    //                 style={child.props.style}
    //                 hoverColor={child.props.hoverColor || this.props.hoverColor}
    //                 onMouseEnter={this.onItemEnter(index)}
    //                 onMouseLeave={this.onItemLeave}
    //                 onClick={child.props.onClick}
    //             >
    //                 {child.props.children}
    //             </ListItem>
    //         );
    //     }
    //     return child;
    // }

    // onListClick = (index: number) => {
    //     return (e: React.MouseEvent<HTMLElement>) => {
    //         e.stopPropagation();
    //         if (this.state.expandedLists.includes(index)){
    //             const expandedLists = [...this.state.expandedLists];
    //             const i = expandedLists.indexOf(index);
    //             expandedLists.splice(i, 1);
    //             this.setState({expandedLists});
    //         } else {
    //             const expandedLists = [...this.state.expandedLists];
    //             expandedLists.push(index);
    //             this.setState({expandedLists});
    //         }
    //     };
    // }

    // setActiveState = (index: number) => {
    //     if (!this.state.expandedLists.includes(index)){
    //         const expandedLists = [...this.state.expandedLists];
    //         expandedLists.push(index);
    //         this.setState({expandedLists});
    //     }
    // }

    // onItemEnter = (index: number) => {
    //     return (e: React.MouseEvent<HTMLElement>) => {
    //         this.setState({hoverItem: index});
    //     };
    // }

    // onItemLeave = () => {
    //     this.setState({hoverItem: -1});
    // }

    // getItemChildren = (
    //     children: Array<React.ReactElement<IExpandedSectionProps> | React.ReactElement<IItemContentProps>>,
    //     expanded: boolean,
    //     index: number
    // ) => {
    //     let content;
    //     let expandedSection;
    //     children.forEach((child) => {
    //         if (child.type === ExpandedSection){
    //             // const expanded = this.state.expandedLists.includes(index) ? true : false;
    //             expandedSection = (
    //                 <ExpandedSection
    //                     className={child.props.className}
    //                     style={child.props.style}
    //                     expanded={expanded}
    //                 >
    //                     {child.props.children}
    //                 </ExpandedSection>
    //             );
    //         } else {
    //             content = child;
    //         }
    //     });
    //     return {content, expandedSection};
    // }
}

export const ListItem: React.StatelessComponent<IListItemProps> = (props) => {

    const {className, /* description, image, */ style, /* title */} = props;

    // const hoverColor = props.hoverColor && props.hovered ? {backgroundColor: props.hoverColor} : {};

    // const getImage = () => {
    //     return (
    //         <div className={styles.image}>
    //             <img src={image}/>
    //         </div>
    //     );
    // };

    // const getTitle = () => {
    //     return title || description ?
    //     (
    //         <div className={`${styles.titleContainer}`}>
    //             {title && <span className={styles.title}>{title}</span>}
    //             {description && <span className={styles.description}>{description}</span>}
    //         </div>
    //     ) : null;
    // };

    // const getContent = () => {
    //     const classes = title || description ? '' : styles.lone;
    //     return props.children &&
    //     (
    //         <div className={`${styles.content} ${classes}`}>
    //             {props.children}
    //         </div>
    //     );
    // };

    // const getDetails = () => {
    //     const classes = !title || !description ? styles.loneImage : '';
    //     return image || title || description ?
    //     (
    //         <div className={`${styles.detailsContainer} ${classes}`}>
    //             {image && getImage()}
    //             {getTitle()}
    //         </div>
    //     ) : null;
    // };

    return(
        <div
            className={`listitem-component ${styles.listItem} ${className}`}
            style={{...style, /* ...hoverColor */}}
            // onMouseEnter={props.onMouseEnter}
            // onMouseLeave={props.onMouseLeave}
            onClick={props.onClick}
        >
            {/* {getDetails()} */}
            {/* {getContent()} */}
            {props.children}
        </div>
    );
};

export interface IExpandedSectionProps {
    id?: string;
    className?: string;
    style?: React.CSSProperties;
    expanded: boolean;
    arrow?: boolean;
    children?: JSX.Element | Element | Element[] | JSX.Element[] | string;
}

export const ExpandedSection: React.StatelessComponent<IExpandedSectionProps> = (props) => {

    const display = props.expanded ? {display: 'block'} : {display: 'none'};

    const displayArrow = props.arrow !== undefined ? props.arrow : true;

    const arrow = () => {
        return displayArrow && (
            <div className={styles.arrow}>
                <div style={{position: 'relative', height: '100%', width: '100%'}}>
                    <div className={styles.innerArrow}/>
                    <div className={styles.borderArrow}/>
                </div>
            </div>
        ) || null;
    };

    return(
        <div
            id={props.id}
            className={`expanded-section-component ${styles.expandedSection} ${props.className}`}
            style={{...display, ...props.style}}
            onClick={(e) => e.stopPropagation()}
        >
            {arrow()}
            {props.children}
        </div>
    );
};

export interface IItemContentProps {
    id?: string;
    className?: string;
    style?: React.CSSProperties;
    children?: JSX.Element | Element | Element[] | JSX.Element[] | string;
}

export const ItemContent: React.StatelessComponent<IItemContentProps> = (props) => {

    return(
        <div
            id={props.id}
            className={`item-content-component ${styles.content} ${props.className}`}
            style={props.style}
        >
            {props.children}
        </div>
    );
};

ListItem.defaultProps = {
    className: '',
    style: {}
};

// export interface IExpandableProps {
//     className?: string;
//     style?: React.CSSProperties;
//     expanded?: boolean;
// }

// export const Expandable: React.StatelessComponent<IExpandableProps> = (props) => {

//     const display = props.expanded ? {display: 'block'} : {display: 'none'};

//     return(
//         <div
//             className={`expanded-section-component ${styles.expandedSection} ${props.className}`}
//             style={{...display, ...props.style}}
//         >
//             {props.children}
//         </div>
//     );
// };
