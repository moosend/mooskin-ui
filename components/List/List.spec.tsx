import * as React from 'react';
import List, {ExpandedSection, ItemContent, ListItem} from './List';

import { mount, shallow } from 'enzyme';

describe('Button', () => {

    test('renders List correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders ListItem correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <ListItem
                className="myClass"
                style={{color: 'blue'}}
            >
                <ItemContent>
                    test
                </ItemContent>
                <ExpandedSection expanded>
                    expanded test
                </ExpandedSection>
            </ListItem>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders with multiple children', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        test
                    </ItemContent>
                    <ExpandedSection expanded arrow={false}>
                        expanded test
                    </ExpandedSection>
                </ListItem>
                <ListItem
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        test
                    </ItemContent>
                    <ExpandedSection expanded={false}>
                        expanded test
                    </ExpandedSection>
                </ListItem>
            </List>
        );

        expect(component).toMatchSnapshot();
    });

    // test('renders ListItems correctly depending on props', () => {

    //     const component = mount(
    //         <List
    //             className="myClass"
    //             style={{color: 'blue'}}
    //             id={'button1'}
    //         >
    //             <ListItem
    //                 description="desc"
    //                 title="title"
    //                 className="myClass"
    //                 style={{color: 'blue'}}
    //             />
    //         </List>
    //     );

    //     expect(component.find('.detailsContainer').length).toBe(2);

    //     expect(component.find('img').length).toBe(0);
    //     expect(component.find('.content').length).toBe(1);

    //     expect(component.find('.title').length).toBe(2);
    //     expect(component.find('.description').length).toBe(2);

    // });

    // test('renders ListItems correctly with different props', () => {

    //     const component = mount(
    //         <List
    //             className="myClass"
    //             style={{color: 'blue'}}
    //             id={'button1'}
    //         >
    //             <ListItem
    //                 image="imageURL"
    //                 className="myClass"
    //                 style={{color: 'blue'}}
    //             >
    //                 <ItemContent>
    //                     Some Content
    //                 </ItemContent>
    //             </ListItem>
    //         </List>
    //     );

    //     expect(component.find('.detailsContainer').length).toBe(1);

    //     expect(component.find('img').length).toBe(1);
    //     expect(component.find('.content').length).toBe(1);

    //     expect(component.find('.title').length).toBe(0);
    //     expect(component.find('.description').length).toBe(0);

    // });

    test('applies the correct classes depending on ListItem content', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        Some Content
                    </ItemContent>
                </ListItem>
            </List>
        );

        expect(component.find('.listitem-component').hasClass('myClass')).toBe(true);
        expect(component.find('.listitem-component').prop('style')).toEqual({color: 'blue'});

    });

    test('create expanded section content correctly', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        Some Content
                    </ItemContent>
                    <ExpandedSection expanded>
                        Some ExpandedSection Content
                    </ExpandedSection>
                </ListItem>
            </List>
        );

        expect(component).toMatchSnapshot();
    });

    test('create expanded section content when proper child is assigned', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem>
                    <ItemContent>
                        Some Content
                    </ItemContent>
                    <ExpandedSection expanded>
                        Some ExpandedSection Content
                    </ExpandedSection>
                </ListItem>
                <ListItem>
                    <ItemContent>
                        Some Content
                    </ItemContent>
                    <ExpandedSection expanded={false}>
                        ExpandedSection content the second
                    </ExpandedSection>
                </ListItem>
            </List>
        );

        expect(component.find('.expanded-section-component').length).toBe(2);
        expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'block'});
        expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'none'});

    });

    // test('render multiple expanded section items and multi items can be active', () => {

    //     const component = mount(
    //         <List
    //             className="myClass"
    //             style={{color: 'blue'}}
    //             id={'button1'}
    //         >
    //             <ListItem>
    //                 <ItemContent>
    //                     Some Content
    //                 </ItemContent>
    //                 <ExpandedSection>
    //                     Some ExpandedSection Content
    //                 </ExpandedSection>
    //             </ListItem>
    //             <ListItem>
    //                 <ItemContent>
    //                     Some Content
    //                 </ItemContent>
    //                 <ExpandedSection>
    //                     ExpandedSection content the second
    //                 </ExpandedSection>
    //             </ListItem>
    //         </List>
    //     );

    //     expect(component.find('.expanded-section-component').length).toBe(2);
    //     expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'none'});
    //     expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'none'});
    //     expect(component.state('activeLists')).toEqual([]);

    //     component.find('.listitem-component').first().simulate('click');

    //     expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'block'});
    //     expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'none'});
    //     expect(component.state('activeLists')).toEqual([0]);

    //     component.find('.listitem-component').last().simulate('click');

    //     expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'block'});
    //     expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'block'});
    //     expect(component.state('activeLists')).toEqual([0, 1]);

    //     component.find('.listitem-component').first().simulate('click');

    //     expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'none'});
    //     expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'block'});
    //     expect(component.state('activeLists')).toEqual([1]);

    //     component.find('.listitem-component').last().simulate('click');

    //     expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'none'});
    //     expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'none'});
    //     expect(component.state('activeLists')).toEqual([]);

    // });

});
