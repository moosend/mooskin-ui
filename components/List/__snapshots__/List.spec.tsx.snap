// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button create expanded section content correctly 1`] = `
<List
  className="myClass"
  id="button1"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className="list-component list myClass"
    id="button1"
    style={
      Object {
        "color": "blue",
      }
    }
  >
    <ListItem
      className="myClass"
      style={
        Object {
          "color": "blue",
        }
      }
    >
      <div
        className="listitem-component listItem myClass"
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <ItemContent>
          <div
            className="item-content-component content undefined"
          >
            Some Content
          </div>
        </ItemContent>
        <ExpandedSection
          expanded={true}
        >
          <div
            className="expanded-section-component expandedSection undefined"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            <div
              className="arrow"
            >
              <div
                style={
                  Object {
                    "height": "100%",
                    "position": "relative",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="innerArrow"
                />
                <div
                  className="borderArrow"
                />
              </div>
            </div>
            Some ExpandedSection Content
          </div>
        </ExpandedSection>
      </div>
    </ListItem>
  </div>
</List>
`;

exports[`Button renders List correctly 1`] = `
<div
  className="list-component list myClass"
  id="button1"
  style={
    Object {
      "color": "blue",
    }
  }
/>
`;

exports[`Button renders ListItem correctly 1`] = `
<div
  className="listitem-component listItem myClass"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <ItemContent>
    test
  </ItemContent>
  <ExpandedSection
    expanded={true}
  >
    expanded test
  </ExpandedSection>
</div>
`;

exports[`Button renders with multiple children 1`] = `
<List
  className="myClass"
  id="button1"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className="list-component list myClass"
    id="button1"
    style={
      Object {
        "color": "blue",
      }
    }
  >
    <ListItem
      className="myClass"
      style={
        Object {
          "color": "blue",
        }
      }
    >
      <div
        className="listitem-component listItem myClass"
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <ItemContent>
          <div
            className="item-content-component content undefined"
          >
            test
          </div>
        </ItemContent>
        <ExpandedSection
          arrow={false}
          expanded={true}
        >
          <div
            className="expanded-section-component expandedSection undefined"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            expanded test
          </div>
        </ExpandedSection>
      </div>
    </ListItem>
    <ListItem
      className="myClass"
      style={
        Object {
          "color": "blue",
        }
      }
    >
      <div
        className="listitem-component listItem myClass"
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <ItemContent>
          <div
            className="item-content-component content undefined"
          >
            test
          </div>
        </ItemContent>
        <ExpandedSection
          expanded={false}
        >
          <div
            className="expanded-section-component expandedSection undefined"
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div
              className="arrow"
            >
              <div
                style={
                  Object {
                    "height": "100%",
                    "position": "relative",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="innerArrow"
                />
                <div
                  className="borderArrow"
                />
              </div>
            </div>
            expanded test
          </div>
        </ExpandedSection>
      </div>
    </ListItem>
  </div>
</List>
`;
